base64_encode.go

```diff
@@ -0,0 +1,47 @@
+package gophplib
+
+import (
+	"encoding/base64"
+	"fmt"
+	"math"
+	"reflect"
+)
+
+// Base64Encode emulates the functionality of PHP 5.6's base64_encode function.
+// For more information, see the [official PHP documentation].
+// In PHP 5.6, an error is triggered if the input size is excessively large due to memory limitations.
+// This Go implementation includes similar checks to emulate PHP's memory limitation conditions.
+// Additionally, this function converts different types of variables to a string, following PHP's dynamic typing approach.
+// After ensuring the memory constraints are met and converting the input to a string,
+// it uses the EncodeToString function from the encoding/base64 package to perform the Base64 encoding.
+// The result is a Base64 encoded string, consistent with PHP's output for the same input.
+// For more detailed information about the EncodeToString function in the package encoding/base64,
+// see the [encoding/base64's EncodeToString documentation]
+//
+// PHP references:
+//   - base64_encode definition:
+//     https://github.com/php/php-src/blob/php-5.6.40/ext/standard/base64.c#L224-L241
+//   - base64_encode implementation:
+//     https://github.com/php/php-src/blob/4b8f72da5dfb201af4e82dee960261d8657e414f/ext/standard/base64.c#L56-L106
+//
+// Test Cases :
+//   - https://github.com/php/php-src/blob/php-5.6.40/ext/standard/tests/url/base64_encode_basic_001.phpt
+//   - https://github.com/php/php-src/blob/php-5.6.40/ext/standard/tests/url/base64_encode_basic_002.phpt
+//   - https://github.com/php/php-src/blob/php-5.6.40/ext/standard/tests/url/base64_encode_error_001.phpt
+//   - https://github.com/php/php-src/blob/php-5.6.40/ext/standard/tests/url/base64_encode_variation_001.phpt
+//
+// [official PHP documentation]: https://www.php.net/manual/en/function.base64-encode
+// [encoding/base64's EncodeToString documentation]: https://pkg.go.dev/encoding/base64#Encoding.EncodeToString
+func Base64Encode(value any) (string, error) {
+	// Convert a value to string
+	characterString, err := zendParseArgAsString(value)
+	if err != nil {
+		return "", fmt.Errorf("unsupported type : %s", reflect.TypeOf(value))
+	}
+
+	if (len(characterString)+2)/3 > math.MaxInt32/4 {
```

Comment:
I think it'd be better why this constraint exists.It would be nice to have a more detailed explanation of why this constraint exists.

I searched it and found that base64 encode convert 3 bytes(24 bits) into 4 bytes(32 bits) ASCII characters, so it should be multiples of 3.

And after it's converted, it could be 133%(4/3) size of original.
So... the size of original data will become 4/3, it should not be `((lengthOfOriginal + 2) / 3) * 4 > INT_MAX`

And it add 2 to `length` becasue if converted data is not multiples of 4, it will add `=` to the end of converted data.

What I found might be wrong, so please just refer it.

Reply:
I referenced your comments to add an explanation about why PHP checks for memory allocation limits during the base64 encoding process, and why this code is necessary in this function.